pytest_bdd/plugin.py

```diff
@@ -104,7 +103,6 @@ def item_key(item):
         else:
             declaration_order = 0
         func, linenum = item.reportinfo()[:2]
-        info = (func, linenum) if linenum is not None else (func, -1)
-        return (info, declaration_order)
+        return (func, linenum if linenum is not None else -1, declaration_order)
```

Comment:
why not "else 0"? "-1" will make items with None suddenly the most important ones.

Reply:
i think -1 is correct. Lint should be for sure on top of everything, this is the only place where we have no line number.

Reply:
pytest-pylint is not the only plugin that defines custom pytest's Items. Anybody can generate arbitrary Items, it's [encouraged](https://docs.pytest.org/en/stable/example/nonpython.html) by pytest itself.
This means that any custom Item that uses None will take the highest priority if pytest-bdd is installed.

I won't comment anymore on this, up to you if you want to fix it or not.